src/SearchParameters/index.js

```diff
@@ -365,9 +365,16 @@ SearchParameters.prototype = {
    * @return {SearchParameters}
    */
   addFacetRefinement : function addFacetRefinement( facet, value ) {
+    if( this.isRefined( facet, value ) ) {
+      return this;
```

Comment:
Shouldn't it return a `this.mutateMe(noop)` instead ? So the return of `addFacetRefinement` is consistenly a new object ?